### ISSUE TITLE
evil-ex-global(i.e. :g) has no ability to achive similar case-sensitivity behavior like vim with 'set ignorecase'/'set noignorecase'

### ISSUE THREAD

**Issue by ogdenwebb** [emacs-evil/evil#1024]
# Issue type

- Bug report

# Environment

Emacs version: GNU Emacs 25.3.1 (x86_64-pc-linux-gnu, GTK+ Version 3.22.19) of 2018-02-12
Operating System: Gentoo Linux
Evil version: Evil version 1.2.13
Evil installation type: MELPA
Graphical/Terminal: GUI version
Tested in a `make emacs` session: no

# Reproduction steps
- Start Emacs
- Write some text in the buffer
```emacs-lisp
baz
bar
Banana
```

- Run `:g/^b/d`

# Expected behavior

```emacs-lisp
;; In vim with :set noignorecase
Banana
```

# Actual behavior

```emacs-lisp
;; The empty buffer.
```
# Further notes

As I can see [here in evil-commands.el](https://github.com/emacs-evil/evil/blob/3bf198796b2d135776143eac854e8d855cd94475/evil-commands.el#L3602) there's nothing related to evil-ex-search-case/evil-ex-substitute-case/any other variable.

**Comment by wasamasa** [comment]
@edkolev I think you should know what exactly is causing this as you've contributed a few PRs recently that improved Ex support.

**Comment by edkolev** [comment]
Sure, I'll look into this.

**Comment by edkolev** [comment]
I've submitted a PR.

On a side note, I like how round the number of this issue is.

**Comment by wasamasa** [comment]
Closed by #1025
